src/main/blackbox/blackbox.c

```diff
@@ -636,7 +636,7 @@ static bool testBlackboxConditionUncached(FlightLogFieldCondition condition)
 #endif
 
     case FLIGHT_LOG_FIELD_CONDITION_RSSI:
-        return rxConfig()->rssi_channel > 0 || feature(FEATURE_RSSI_ADC);
+        return getRSSISource() != RSSI_SOURCE_NONE;
```

Comment:
This is potentially unsafe. We assume that `blackboxStart()` will be always called after `rxInit()`. This is probably true at the moment, but may change if we decide to start logging some boot information to BB before arming. Can you add a comment that this is dependent on `rxInit()` being called before starting the logging?

Reply:
Done. I had gone with the assumption that blackbox logging only can start after the arming checks have succeeded, which would include the `rxInit()` as part of the FAILSAFE_RX arming condition but you're right there's no guarantee if the user force arms.